File: Java/Akash-212/605_Can_Place_Flowers.java
class Solution {
    public boolean canPlaceFlowers(int[] flowerbed, int n) {
      int count = 0;
        int len = flowerbed.length;
        for(int i=0;i<len;i++){
            if(flowerbed[i]==0){
                if((i==0 || flowerbed[i-1]==0)&&(i==len-1 ||flowerbed[i+1]==0)){
                flowerbed[i] = 1;
                count++;
                }
            }
        }
        
        return n<=count;
    }

}

File: CPP/RathodTej/605. Can Place Flowers.cpp
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {

        if (n == 0)
            return true;

        int l = flowerbed.size();
        int count = 0;

        for (int i = 0; i < l; i++) {

            if (flowerbed[i] == 0) {

                int leftEmpty = (i == 0) || (flowerbed[i - 1] == 0);
                int rightEmpty = (i == l - 1) || (flowerbed[i + 1] == 0);

                if (leftEmpty && rightEmpty) {
                    flowerbed[i] = 1;
                    count++;

                    if (count >= n)
                        return true;
                }
            }
        }

        return false;
    }
};
